examples/platforms/posix/radio.c

```diff
@@ -708,45 +756,102 @@ void radioProcessFrame(otInstance *aInstance)
 
 void otPlatRadioEnableSrcMatch(otInstance *aInstance, bool aEnable)
 {
+    sSrcMatchEnabled = aEnable;
     (void)aInstance;
-    (void)aEnable;
 }
 
 otError otPlatRadioAddSrcMatchShortEntry(otInstance *aInstance, const uint16_t aShortAddress)
 {
+    otError error = OT_ERROR_NONE;
+    otEXPECT_ACTION(sShortAddressMatchTableCount < sizeof(sShortAddressMatchTable) / sizeof(uint16_t),
+                    error = OT_ERROR_NO_BUFS);
+
+    for (uint8_t i = 0; i < sShortAddressMatchTableCount; ++i)
+    {
+        otEXPECT_ACTION(sShortAddressMatchTable[i] != aShortAddress, error = OT_ERROR_DUPLICATED);
+    }
+
+    sShortAddressMatchTable[sShortAddressMatchTableCount++] = aShortAddress;
+
+exit:
     (void)aInstance;
-    (void)aShortAddress;
-    return OT_ERROR_NONE;
+    return error;
 }
 
 otError otPlatRadioAddSrcMatchExtEntry(otInstance *aInstance, const otExtAddress *aExtAddress)
 {
+    otError error = OT_ERROR_NONE;
+
+    otEXPECT_ACTION(sExtAddressMatchTableCount < sizeof(sExtAddressMatchTable) / sizeof(uint16_t),
+                    error = OT_ERROR_NO_BUFS);
+
+    for (uint8_t i = 0; i < sExtAddressMatchTableCount; ++i)
+    {
+        otEXPECT_ACTION(memcmp(&sExtAddressMatchTable[i], aExtAddress, sizeof(otExtAddress)),
+                        error = OT_ERROR_DUPLICATED);
+    }
+
+    sExtAddressMatchTable[sExtAddressMatchTableCount++] = *aExtAddress;
+
+exit:
     (void)aInstance;
-    (void)aExtAddress;
-    return OT_ERROR_NONE;
+    return error;
 }
 
 otError otPlatRadioClearSrcMatchShortEntry(otInstance *aInstance, const uint16_t aShortAddress)
 {
+    otError error = OT_ERROR_NONE;
+    otEXPECT_ACTION(sShortAddressMatchTableCount > 0, error = OT_ERROR_NOT_FOUND);
+
+    for (uint8_t i = 0; i < sShortAddressMatchTableCount; ++i)
+    {
+        if (sShortAddressMatchTable[i] == aShortAddress)
+        {
+            sShortAddressMatchTable[i] = sShortAddressMatchTable[--sShortAddressMatchTableCount];
+            error = OT_ERROR_NONE;
+            goto exit;
+        }
+    }
+
+    error = OT_ERROR_NOT_FOUND;
+
+exit:
     (void)aInstance;
-    (void)aShortAddress;
-    return OT_ERROR_NONE;
+    return error;
 }
 
 otError otPlatRadioClearSrcMatchExtEntry(otInstance *aInstance, const otExtAddress *aExtAddress)
 {
+    otError error = OT_ERROR_NONE;
```

Comment:
initialized as `OT_ERROR_NOT_FOUND` and remove Line#839 and `error = OT_ERROR_NOT_FOUND` in Line#827?